client/settings/general-settings-section/payment-method-checkbox.js

```diff
@@ -74,12 +61,6 @@ const PaymentMethodCheckbox = ( {
 		setEnabledPaymentMethods(
 			enabledPaymentMethods.filter( ( m ) => m !== id )
 		);
-
-		// In legacy mode (UPE disabled), Stripe refers to the card payment method.
-		// So if the card payment method is disabled, Stripe should be disabled.
-		if ( id === 'card' && ! isUpeEnabled ) {
-			setIsStripeEnabled( false );
-		}
```

Comment:
This change and the one above it are the core of this fix. This code would enable/disable the gateway when you enabled/disabled cards.